purchases/src/main/kotlin/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -510,6 +513,23 @@ internal class Backend(
         PostReceiptErrorHandlingBehavior.SHOULD_BE_CONSUMED
     }
 
+    private fun <S, E> MutableMap<BackgroundAwareCallbackCacheKey, MutableList<Pair<S, E>>>.addBackgroundAwareCallback(
+        call: Dispatcher.AsyncCall,
+        dispatcher: Dispatcher,
+        cacheKey: BackgroundAwareCallbackCacheKey,
+        functions: Pair<S, E>,
+        delay: Delay = Delay.NONE,
+    ) {
+        val foregroundCacheKey = Pair(cacheKey.first, false)
+        val cacheKeyToUse = if (cacheKey.second && callbacks.containsKey(foregroundCacheKey)) {
+            foregroundCacheKey
+        } else {
+            cacheKey
+        }
+        addCallback(call, dispatcher, cacheKeyToUse, functions, delay)
+        moveBackgroundedCallbacksToForegroundedCallbacksIfNeeded(this, cacheKey.first, cacheKey.second)
```

Comment:
It's only necessary to do this if this new callback being added is unjittered right?

Reply:
Yes, that's correct. We do that check inside the method, but we might be able to move it here.

Reply:
Removed that function and moved it to the main one, so it's clearer in https://github.com/RevenueCat/purchases-android/pull/1204/commits/6aa761f05530ab99af09918c162165809d41a83e